src/System.Net.Security/tests/FunctionalTests/SslStreamAlertsTest.cs

```diff
@@ -0,0 +1,168 @@
+            {
+                Task serverAuth = server.AuthenticateAsServerAsync(certificate);
+                await client.AuthenticateAsClientAsync(certificate.GetNameInfo(X509NameType.SimpleName, false));
+
+                byte[] buffer = new byte[1024];
+
+                // Schannel semantics require that Decrypt is called to receive an alert.
+                await client.WriteAsync(buffer, 0, buffer.Length);
+                var exception = await Assert.ThrowsAsync<IOException>(() => client.ReadAsync(buffer, 0, buffer.Length));
+
+                Assert.IsType<Win32Exception>(exception.InnerException);
+                var win32ex = (Win32Exception)exception.InnerException;
+
+                // The Schannel HResults for each alert are documented here: 
+                // https://msdn.microsoft.com/en-us/library/windows/desktop/dd721886(v=vs.85).aspx
+                Assert.Equal(SEC_E_ILLEGAL_MESSAGE, (uint)win32ex.NativeErrorCode);
+
+                await Assert.ThrowsAsync<AuthenticationException>(() => serverAuth);
+
+                await Assert.ThrowsAsync<AuthenticationException>(() => server.WriteAsync(buffer, 0, buffer.Length));
+                await Assert.ThrowsAsync<AuthenticationException>(() => server.ReadAsync(buffer, 0, buffer.Length));
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public async Task SslStream_StreamToStream_ServerInitiatedCloseNotify_Ok()
+        {
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var serverStream = new VirtualNetworkStream(network, isServer: true))
+            using (var client = new SslStream(clientStream, true, AllowAnyServerCertificate))
+            using (var server = new SslStream(serverStream))
+            using (X509Certificate2 certificate = Configuration.Certificates.GetServerCertificate())
+            {
+                var handshake = new Task[2];
+
+                handshake[0] = server.AuthenticateAsServerAsync(certificate);
+                handshake[1] = client.AuthenticateAsClientAsync(certificate.GetNameInfo(X509NameType.SimpleName, false));
+
+                await Task.WhenAll(handshake).TimeoutAfter(TestConfiguration.PassingTestTimeoutMilliseconds);
+
+                var readBuffer = new byte[1024];
+
+                await server.ShutdownAsync();
+                int bytesRead = await client.ReadAsync(readBuffer, 0, readBuffer.Length);
+                // close_notify received by the client.
+                Assert.Equal(0, bytesRead);
+
+                await client.ShutdownAsync();
+                bytesRead = await server.ReadAsync(readBuffer, 0, readBuffer.Length);
+                // close_notify received by the server.
+                Assert.Equal(0, bytesRead);
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
```

Comment:
`PlatformID` has been renamed `TestPlatforms`. It would be good if it was updated here before merging. See #12284.